src/Microsoft.TestPlatform.Extensions.TrxLogger/TrxLogger.cs

```diff
@@ -389,9 +393,17 @@ internal void TestRunCompleteHandler(object sender, TestRunCompleteEventArgs e)
 
             helper.SaveObject(runSummary, rootElement, "ResultSummary", parameters);
 
-            //Save results to Trx file
-            this.DeriveTrxFilePath();
-            this.PopulateTrxFile(this.trxFilePath, rootElement);
+            EqtAssert.IsTrue(createLogFileMutex.WaitOne(MutexTimeout), "Cannot acquire the global file population mutex.");
```

Comment:
Could we instead of using mutex to lock the system for the whole time it populates the file just simply create the file with FileMode.CreateNew instead of FileMode.Create, and handle the IOException by generating a new timestamp?

Reply:
I can't see any downsides of using a Mutex, it actually feels cleaner then starting the operation, handling the exception and trying again. Am I missing something?

Reply:
My thinking is this:
1) Mutex is probably not behaving perfectly the same cross platform. So making sure this works on RedHat, MacOS, Linux, Windows and Windows ARM might be unnecessarily difficult.
2) the conflict only happens occasionally, but we will lock every time 
3) in case we get into conflict one process has to wait for the whole time the other is generating the whole file, not just reserving the name
( 4) there are not Mutexes in the solution so far )

All in all the solution seems overcomplicated for something that can be done reactively when the writing actually fails.  That solution is also much easier to test.